swupd-inspector/archive.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"archive/tar"
+	"bytes"
+	"compress/bzip2"
+	"compress/gzip"
+	"fmt"
+	"io"
+
+	"github.com/clearlinux/mixer-tools/swupd"
+)
+
+// TODO: Evaluate where to put this, partially duplicate with code in packs.
+
+type compressedTarReader struct {
+	*tar.Reader
+	compressionCloser io.Closer
+}
+
+func (ctr *compressedTarReader) Close() error {
+	if ctr.compressionCloser != nil {
+		return ctr.compressionCloser.Close()
+	}
+	return nil
+}
+
+// Compression algorithms have "magic" bytes in the beginning of the file to identify them.
+var (
```

Comment:
These are definitely already defined in the swupd package. Those should be exported if you need to use them here.

Reply:
Ditto.